mzLib/Omics/IBioPolymerWithSetMods.cs

```diff
@@ -163,5 +163,39 @@ public static Dictionary<int, Modification> GetModificationDictionaryFromFullSeq
         /// <returns></returns>
         public static List<Modification> GetModificationsFromFullSequence(string fullSequence,
             Dictionary<string, Modification> allModsKnown) => [.. GetModificationDictionaryFromFullSequence(fullSequence, allModsKnown).Values];
+
+        /// <summary>
+        /// Determines the full sequence of a BioPolymerWithSetMods from its base sequence and modifications
+        /// </summary>
+        public static string DetermineFullSequence(string baseSequence, IDictionary<int, Modification> allModsOneIsNterminus)
+        {
+            // start string builder with initial capacity to avoid resizing costs. 
+            var subSequence = new StringBuilder(baseSequence.Length + allModsOneIsNterminus.Count * 30);
```

Comment:
maybe take this out to 60 instead of 30. for example:

Proline pyrrole to pyrrolidine six member ring on P is 51.

Reply:
Made this change